server/src/com/cloud/template/TemplateAdapterBase.java

```diff
@@ -190,11 +190,12 @@ public TemplateProfile prepare(boolean isIso, long userId, String name, String d
 
         // check whether owner can create public templates
         boolean allowPublicUserTemplates = TemplateManager.AllowPublicUserTemplates.valueIn(templateOwner.getId());
+        boolean isAdmin = _accountMgr.isAdmin(templateOwner.getId());
```

Comment:
@Slair1 This change is what allows for Domain Admins to upload featured/public templates. `isAdmin()` will check for root or domain admin. I renamed the other variables that were previously `isAdmin` to `isRootAdmin` to be clearer on what the code was actually doing before my changes.

Reply:
@nlgordon Can a Domain Admin create a Featured template that is only visible and featured in their Domain?

Reply:
Yes. With our new setting turned on the template will only be visible in their domain. With the new setting turned off it will be visible globally. It would also be easy to make it so that domain admins can only upload featured templates if the new `restrict.public.template.access.to.domain`setting is turned on.

Reply:
I did some more testing, and I misspoke earlier. Public was working, but not featured for domain admins. New commit fixes that and ties the domain admin ability to upload public/featured.